scipy/stats/stats.py

```diff
@@ -601,7 +623,7 @@ def hmean(a, axis=0, dtype=None):
         raise ValueError("Harmonic mean only defined if all elements greater than zero")
 
 
-def mode(a, axis=0):
+def mode(a, axis=0, nan_policy='propagate'):
```

Comment:
If you just remove the special casing of ``nan_policy == 'propagate':``, then you should get the following:

   >>> stats.mode([np.nan, np.nan, 3])
   nan
   >>> stats.mode([np.nan, 3, 3])
   3